sdk/js/index.ts

```diff
@@ -205,4 +205,44 @@ export default class StreamManager {
       onLogs,
     });
   }
+
+  async cancel(
+    streamId: number,
+    reason: string,
+    creator: Address
+  ): Promise<bigint> {
+    if (!this.walletClient)
+      throw new Error("Error cancelling stream: wallet client is not set");
+    const result = await this.walletClient.writeContract({
+      chain: null,
+      address: this.address,
+      abi: StreamManagerContractType.abi as Abi,
+      functionName: "cancel_stream",
+      args: [streamId, reason, creator],
+      account: creator,
+    });
+    if (result === undefined)
+      throw new Error("Error while processing transactions");
+    return BigInt(result);
+  }
+
+  async update(
+    creator: Address,
+    streamId: number,
+    amount: number
+  ): Promise<bigint> {
+    if (!this.walletClient)
+      throw new Error("Error funding stream: wallet client is not set");
+    const timeLeft = await this.walletClient.writeContract({
```

Comment:
question: Should we be handling token approvals (check & set) in the SDK?  This will fail if the proper approval hasn't been set.  Same concern with `create()` which might be out of scope of this PR.

Reply:
I think the answer is yes, but not in the SDK (which appears to just be a thin contract wrapper).  Here's how it's done in the `CreateStream` component:

https://github.com/ApeWorX/ApePay/blob/61ad4b17996723b5a806aa21400a33b32c8d28af/ui/lib/CreateStream.tsx#L396-L403

Without taking this step to set the ERC20 allowance (if necessary), the call to `add_funds()` will fail.  Let me know if you have any questions.

Reply:
The add value to stream functionality should be upstream to the SDK